scripts/ingests/utils.py

```diff
@@ -536,9 +542,21 @@ def ingest_sources(db, sources, ras, decs, references, epochs = None, equinoxes
             db.Sources.insert().execute(source_data)
             n_added += 1
         except sqlalchemy.exc.IntegrityError as err:
-            print("SIMPLE ERROR: Discovery reference may not exist in the Publications table. Add it with add_publication function. ")
-            with disable_exception_traceback():
-                raise
+            # try reference without last letter e.g.Smit04 instead of Smit04a
+            if source_data[0]['reference'][:-1] == 'a' or 'b':
```

Comment:
This is always ```True``` (try ```bool('b')```, also I'm assuming you're expected a string, e.g. 'asdf01a', but are using a slice which would return in this case 'asdf01'; sounds like what you're looking for is
```suggestion
            if source_data[0]['reference'][-1] in ('a', 'b'):
```